File: Store.BusinessLogic/Models/Users/UserModelItem.cs
﻿using Store.BusinessLogic.Common.Constants;
using Store.BusinessLogic.Models.Base;

namespace Store.BusinessLogic.Models.Users
{
    public class UserModelItem : BaseModel
    {
        public long Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public bool IsLocked { get; set; }
        public string Avatar { get; set; } = Constants.Images.DefaultUserAvatar;
        public string Role { get; set; }
    }
}

File: Store.View/src/app/shared/models/user/user.model.item.ts
import { BaseModel } from 'src/app/shared/models/base/base.model';

export class UserModelItem extends BaseModel {
    public id?: number;
    public firstName?: string;
    public lastName?: string;
    public email?: string;
    public password?: string;
    public newPassword?: string;
    public isLocked?: boolean;
    public avatar?: string;
    public role?: string;
}
